doc/acl/egress_outer_dscp_change_table.md

```diff
@@ -0,0 +1,505 @@
+     * |  MATCH_L4_DST_PORT     |              |            |
+     * |  MATCH_DSCP            |              |            |
+     * |----------------------------------------------------|
+
+     * EGR_SET_DSCP
+     * |----------------------------------------------------|
+     * |                        |              |            |
+     * |    Match Type          |   Actions    |    Bind    |
+     * |                        |              |   points   |
+     * |----------------------------------------------------|
+     * |  MATCH_METADATA        | SET_DSCP     |    PORT    |
+     * |                        |              |    LAG     |
+     * |----------------------------------------------------|
+     */
+
+
+##### Metadata Management
+
+Since not all platforms support metadata matching and action, the implementation would use SAI capability check to see if metadata action and match attributes are supported. In addition, the range of the metadata field would be checked by 'SAI_SWITCH_ATTR_ACL_USER_META_DATA_RANGE'.
+The orcahagent would check if a DSCP action value has an assigned metadata value, thereby reusing the same value and EGR_SET_DSCP entry. If a DSCP value has no associated metadata value, a new one would be allcoated along with EGR_SET_DSCP entry.
+Since these metadata values would be reused accross differnt ACLs and for both V4 and V6 type entries, Each metadata, DSCP combition would be reference counted.
+During the processing of an ACL, if metadata allocation fails for an entry, the orchagent would return failure for that and subsequent entries and the user would have to remove the complete ACL.
+ 
+
+##### EGR_SET_DSCP Management
+
+The EGR_SET_DSCP table would be created and managed by the orchagent internally. Only a single instance of this table would be created upon creation of UNDERLAY_SET_DSCP/V6 table and subsequent tables of type UNDERLAY_SET_DSCP/V6 would reuse this instance. By employing ref-counting this table would be retained until the last user of this table is removed. Entries in this table would also be ref-counted as mnetioned in above section.
+The interface association of this table can be implemented in two ways. 
+
+  1) EGR_SET_DSCP association with all dataplane interfaces. This approach is resource intensive and would require changes in portsOrch to accomodate exposing all ports to the AclOrch. However this allows us to decouple the ingress stage table port association from EGR_SET_DSCP.
+  2) EGR_SET_DSCP be associated with a superset of all interfaces which are associated with tables referencing it. e.g. if UNDERLAY_SET_DSCP is associated with interfaces [a,b,c] and UNDERLAY_SET_DSCPV6 is associated with [c,d,e] then EGR_SET_DSCP would be associated with [a,b,c,d,e]. This gives the user of the feature the choice to limit port association when needed. Based on common use case pattern, the tables would almost always be associated with all data-plane interfaces. 
```

Comment:
The suggested use-case was that these rules are going to be common for all the ports. In that case, it would be easier to implement this with bindpoint as SWITCH.

Reply:
agreed. added to the addendum and would be implemted in the next iteration.